src/app/modules/admin-page/admin-page.component.ts

```diff
@@ -11,38 +11,67 @@ import { FormBuilder, FormGroup, Validators, FormControl } from '@angular/forms'
   styleUrls: ['./admin-page.component.scss']
 })
 export class AdminPageComponent implements OnInit {
+  @Input() complex;
+  @Input() name;
+  @Input() half;
+  @Input() type;
+  @Input() todaymenu;
+  @Input() price;
+  @Input() weight;
+  @Input() img;
+  @Input() info;
+  @Input() id;
+
   public addDishForm: FormGroup;
   public defaultImage: string = 'assets/pictures/placeholder.jpg';
 
   public filterProp: filterProp[] = PROPERTIES;
   public dish: Dish = {
     complex: '',
+    half: false,
     name: '',
     type: '',
     todaymenu: false,
-    price: '',
-    weight: '',
+    price: 0,
+    weight: 0,
     img: '',
     info: ''
   };
-
-  constructor(private _fbs: FirebaseService, private _formBuilder: FormBuilder) {}
+  constructor(private _fbs: FirebaseService, private _formBuilder: FormBuilder) {
+  }
 
   public ngOnInit(): void {
+    this.dish.id = this.id               || this.dish.id;
+    this.dish.complex = this.complex     || this.dish.complex;
+    this.dish.half = this.half           || this.dish.half;
+    this.dish.name = this.name           || this.dish.name;
+    this.dish.type = this.type           || this.dish.type;
+    this.dish.todaymenu = this.todaymenu || this.dish.todaymenu;
+    this.dish.price = this.price         || this.dish.price;
+    this.dish.weight = this.weight       || this.dish.weight;
+    this.dish.img = this.img             || this.dish.img;
+    this.dish.info = this.info           || this.dish.info;
```

Comment:
Why do you need it?

Reply:
If we have dish from input we use that info in other case existing data